File: Selenium/Activities/Activity9.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.firefox.service import Service as FirefoxService

serv = FirefoxService(GeckoDriverManager().install())

with webdriver.Firefox(service=serv) as driver:
    driver.get("https://training-support.net/selenium/ajax")
    print("Title is: ", driver.title)
    wait = WebDriverWait(driver, 10)
    driver.find_element(By.XPATH, "//button[text()='Change Content']").click()
    wait.until(expected_conditions.text_to_be_present_in_element((By.XPATH, "//div/h1"), "HELLO!"))
    print("Text in h1 tag: ", driver.find_element(By.XPATH, "//div/h1").text)
    wait.until(expected_conditions.text_to_be_present_in_element((By.XPATH, "//div/h3"), "I'm late!"))
    print("Text in h3 tag: ", driver.find_element(By.XPATH, "//div/h3").text)
    driver.quit()

File: Selenium/Activities/Activity9.java
package Activities;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedCondition;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;
import static org.openqa.selenium.support.ui.ExpectedConditions.*;

public class Activity9 {
    public static void main(String[] args) throws InterruptedException {
        WebDriverManager.firefoxdriver().setup();
        WebDriver newDriver = new FirefoxDriver();
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverWait wait = new WebDriverWait(newDriver, Duration.ofSeconds(10));
        newDriver.get("https://training-support.net/selenium/ajax");
        System.out.println("Title is: "+newDriver.getTitle());
        WebElement buttonEle = newDriver.findElement(By.className("violet"));
        buttonEle.click();
        Thread.sleep(300);
        WebElement helloEle = newDriver.findElement(By.xpath("//div[@id='ajax-content']/h1"));
        wait.until(ExpectedConditions.textToBePresentInElement(helloEle,"HELLO!"));
        System.out.println(helloEle.getText());
        WebElement lateEle = newDriver.findElement(By.tagName("h3"));
        System.out.println(lateEle.getText());
        wait.until(ExpectedConditions.textToBePresentInElementLocated(By.tagName("h3"), "I'm late!"));
        System.out.println(newDriver.findElement(By.tagName("h3")).getText());
        Thread.sleep(300);
        newDriver.quit();
    }
}


